src/ComboboxInput/__stories__/ComboboxInput.stories.js

```diff
@@ -3,41 +3,43 @@ import Menu from '../../Menu/Menu';
 import React from 'react';
 import { storiesOf } from '@storybook/react';
 import {
+    boolean,
+    select,
     withKnobs
 } from '@storybook/addon-knobs';
 
+const menu = (<Menu>
+    <Menu.List>
+        <Menu.Item url='#'>Pear</Menu.Item>
+        <Menu.Item url='#'>Strawberry</Menu.Item>
+        <Menu.Item url='#'>Raspberry</Menu.Item>
+        <Menu.Item isLink url='#'>
+            + New Item
+        </Menu.Item>
+    </Menu.List>
+</Menu>);
+
+const createProps = (overrides) => ({
+    compact: boolean('compact', false),
+    menu: menu,
+    state: select('Validation State', {
+        'default': null,
+        'valid': 'valid',
+        'invalid': 'invalid',
+        'information': 'information',
+        'warning': 'warning'
+    }),
+    placeholder: 'Select a Fruit',
+    ...overrides
+});
+
 storiesOf('Components|ComboboxInput', module)
     .addDecorator(withKnobs)
     .add('Default', () => (
-        <ComboboxInput
-            menu={
-                <Menu>
-                    <Menu.List>
-                        <Menu.Item url='#'>Pear</Menu.Item>
-                        <Menu.Item url='#'>Strawberry</Menu.Item>
-                        <Menu.Item url='#'>Raspberry</Menu.Item>
-                        <Menu.Item isLink url='#'>
-                            + New Item
-                        </Menu.Item>
-                    </Menu.List>
-                </Menu>
-            }
-            placeholder='Select Fruit' />
+        <ComboboxInput {...createProps()} />
     ))
     .add('disable styles', () => (
         <ComboboxInput
-            disableStyles
-            menu={
-                <Menu>
-                    <Menu.List>
-                        <Menu.Item url='#'>Pear</Menu.Item>
-                        <Menu.Item url='#'>Strawberry</Menu.Item>
-                        <Menu.Item url='#'>Raspberry</Menu.Item>
-                        <Menu.Item isLink url='#'>
-                            + New Item
-                        </Menu.Item>
-                    </Menu.List>
-                </Menu>
-            }
-            placeholder='Select Fruit' />
+            {...createProps()}
```

Comment:
Going forward, it would be good to have one "default" story for each component that contains all the storybook knobs, but all additional stories should probably disable knobs as the stories really become specific use cases with all the appropriate props set and shouldn't be "played with" anymore.  There are a few exceptions to this, yes, but it would be good if that was the general pattern.  That said, not part of this PR.

Reply:
Agree! I started to do that... then realized it's soooo out of scope. What do you think about a story in the backlog to 1) upgrade storybook to use the "new" pattern we've established in floorplans 2) setup the stories consistently for the components?

Reply:
Yeah, that sounds like a good plan.